programs/uxd/src/instructions/mercurial/collect_profit_of_mercurial_vault_depository.rs

```diff
@@ -0,0 +1,298 @@
+use crate::error::UxdError;
+use crate::mercurial_utils;
+use crate::validate_is_program_frozen;
+use crate::Controller;
+use crate::MercurialVaultDepository;
+use crate::CONTROLLER_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_LP_TOKEN_VAULT_NAMESPACE;
+use crate::MERCURIAL_VAULT_DEPOSITORY_NAMESPACE;
+use anchor_lang::prelude::*;
+use anchor_spl::token::Mint;
+use anchor_spl::token::Token;
+use anchor_spl::token::TokenAccount;
+use fixed::types::I80F48;
+
+#[derive(Accounts)]
+pub struct CollectProfitOfMercurialVaultDepository<'info> {
+    /// #1
+    #[account(mut)]
+    pub payer: Signer<'info>,
+
+    /// #2
+    #[account(
+        mut,
+        seeds = [CONTROLLER_NAMESPACE],
+        bump = controller.load()?.bump,
+        constraint = controller.load()?.registered_mercurial_vault_depositories.contains(&depository.key()) @UxdError::InvalidDepository,
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #3
+    #[account(
+        mut,
+        seeds = [MERCURIAL_VAULT_DEPOSITORY_NAMESPACE, depository.load()?.mercurial_vault.key().as_ref(), depository.load()?.collateral_mint.as_ref()],
+        bump = depository.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+        has_one = mercurial_vault @UxdError::InvalidMercurialVault,
+        has_one = collateral_mint @UxdError::InvalidCollateralMint,
+        has_one = mercurial_vault_lp_mint @UxdError::InvalidMercurialVaultLpMint,
+        has_one = profits_beneficiary_collateral @UxdError::InvalidProfitsBeneficiaryCollateral,
+        constraint = depository.load()?.lp_token_vault == depository_lp_token_vault.key() @UxdError::InvalidDepositoryLpTokenVault,
+    )]
+    pub depository: AccountLoader<'info, MercurialVaultDepository>,
+
+    /// #4
+    pub collateral_mint: Box<Account<'info, Mint>>,
+
+    /// #5
+    #[account(
+        mut,
+        constraint = profits_beneficiary_collateral.mint == depository.load()?.collateral_mint @UxdError::InvalidCollateralMint,
+    )]
+    pub profits_beneficiary_collateral: Box<Account<'info, TokenAccount>>,
+
+    /// #6
+    /// Token account holding the LP tokens minted by depositing collateral on mercurial vault
+    #[account(
+        mut,
+        seeds = [MERCURIAL_VAULT_DEPOSITORY_LP_TOKEN_VAULT_NAMESPACE, mercurial_vault.key().as_ref(), collateral_mint.key().as_ref()],
```

Comment:
nit: we dont need to specify the seeds here if we are just using the account address specified in the depository's state?